scripts/preprocess_spec.py

```diff
@@ -30,24 +39,67 @@ def _to_camel_case(s):
     return ''.join(_title(y) for y in s.split("_"))
 
 
-def iterate_through_operations(spec, func):
+def apply_funcs_to_spec(spec, func, *params):
+    """Apply fun to each operation in the spec.
+
+    First two parameters of func will be operation and it's parent, and the
+    rest of parameters would be anything passed after func to this function.
+    """
     for k, v in spec['paths'].iteritems():
         for op in _ops:
-            if op in v:
-                func(v[op])
+            if op not in v:
+                continue
+            if func(v[op], v, *params):
+                del v[op]
+
+
+def _has_property(prop_list, property_name):
+    for prop in prop_list:
+        if prop["name"] == property_name:
+            return True
+
+
+def remove_watch_operations(op, parent, operation_ids):
+    op_id = op['operationId']
+    if not op_id.startswith(WATCH_OP_PREFIX):
+        return
+    list_id = (LIST_OP_PREFIX +
+               op_id.replace(WATCH_OP_SUFFIX, "")[len(WATCH_OP_PREFIX):])
+    if list_id not in operation_ids:
+        print "Cannot find %s" % list_id
+        return
+    list_op = operation_ids[list_id]
+    params = []
+    if 'parameters' in list_op:
+        params += list_op['parameters']
+    if 'parameters' in parent:
+        params += parent['parameters']
+    if not _has_property(params, WATCH_QUERY_PARAM_NAME):
+        print "%s has no watch query param" % list_id
+        return
+    return True
+
+
+def strip_tags_from_operation_id(operation, _):
+    operation_id = operation['operationId']
+    for t in operation['tags']:
+        operation_id = operation_id.replace(_to_camel_case(t), '')
```

Comment:
(No action required)

``to_camel_case`` is only referenced here, consider inlining.